h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ParquetParser.java

```diff
@@ -277,7 +277,10 @@ private static ParquetPreviewParseWriter readFirstRecords(ParquetParseSetup init
           types[i] = OriginalType.TIMESTAMP_MILLIS.equals(parquetType.getOriginalType()) ? Vec.T_TIME : Vec.T_NUM;
           break;
         default:
-          types[i] = Vec.T_BAD;
+          if (OriginalType.DECIMAL.equals(parquetType.getOriginalType())) {
```

Comment:
what is the getPrimitiveTypeName value here? should we rather convert the whole swithc to if-else if that is not usable?

Reply:
good point, @honzasterba please take another look